pkg/ottl/ottlfuncs/func_delete_matching_keys.go

```diff
@@ -33,18 +33,34 @@ func createDeleteMatchingKeysFunction[K any](_ ottl.FunctionContext, oArgs ottl.
 	return deleteMatchingKeys(args.Target, args.Pattern)
 }
 
-func deleteMatchingKeys[K any](target ottl.PMapGetSetter[K], pattern string) (ottl.ExprFunc[K], error) {
-	compiledPattern, err := regexp.Compile(pattern)
-	if err != nil {
-		return nil, fmt.Errorf("the regex pattern supplied to delete_matching_keys is not a valid pattern: %w", err)
+func deleteMatchingKeys[K any](target ottl.PMapGetSetter[K], pattern ottl.StringGetter[K]) (ottl.ExprFunc[K], error) {
+	literalPattern, ok := ottl.GetLiteralValue(pattern)
+	var compiledPattern *regexp.Regexp
+	var err error
+	if ok {
+		compiledPattern, err = regexp.Compile(literalPattern)
+		if err != nil {
+			return nil, fmt.Errorf("the regex pattern supplied to delete_matching_keys is not a valid pattern: %w", err)
+		}
 	}
 	return func(ctx context.Context, tCtx K) (any, error) {
+		cp := compiledPattern
+		if cp == nil {
+			patternVal, err := pattern.Get(ctx, tCtx)
+			if err != nil {
+				return nil, err
+			}
+			cp, err = regexp.Compile(patternVal)
+			if err != nil {
+				return nil, fmt.Errorf("the regex pattern supplied to delete_matching_keys is not a valid pattern: %w", err)
```

Comment:
```suggestion
				return nil, fmt.Errorf("the regex pattern supplied to delete_matching_keys %q is not a valid pattern: %w", patternVal, err)

```

WDYT? I believe It would be useful for troubleshooting considering the pattern can now be dynamic. We could also create a new function for compiling the regex, so we don't need to repeat this same pattern twice.

Reply:
Agreed, the idea of having a function for that was mainly unifying the error's message. The constant you added works as well. Thanks!